tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -258,6 +258,91 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
         }
     }
 
+#if _HAS_CXX23
+    using ranges::const_iterator_t, ranges::const_sentinel_t, ranges::cbegin, ranges::cend;
+
+    // Validate view_interface::cbegin
+    STATIC_ASSERT(CanMemberCBegin<R>);
+    STATIC_ASSERT(CanMemberCBegin<const R> == ranges::range<const R>
+                  && indirect_unary_predicate<const Pred, const_iterator_t<const V>>);
+    if (forward_range<V>) { // intentionally not if constexpr
+        const same_as<const_iterator_t<R>> auto i = r.cbegin();
+        if (!is_empty) {
+            assert(*i == *cbegin(expected));
+            assert(*r.cbegin() == *cbegin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto r2                                    = r;
+            const same_as<const_iterator_t<R>> auto i2 = r2.cbegin();
+            if (!is_empty) {
+                assert(*i2 == *i);
+                assert(*r2.cbegin() == *i2);
+                assert(*r2.cbegin() == *i2);
+            }
+        }
+
+        const same_as<const_iterator_t<const R>> auto ic = as_const(r).cbegin();
+        if (!is_empty) {
+            assert(*ic == *cbegin(expected));
+            assert(*as_const(r).cbegin() == *cbegin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto rc2                                          = as_const(r);
+            const same_as<const_iterator_t<const R>> auto ic2 = rc2.cbegin();
+            if (!is_empty) {
+                assert(*ic2 == *ic);
+                assert(*rc2.cbegin() == *ic2);
+            }
+        }
+    }
+
+    // Validate view_interface::cend
+    STATIC_ASSERT(CanMemberCEnd<R>);
+    STATIC_ASSERT(CanMemberCEnd<const R> == ranges::range<const R>
+                  && indirect_unary_predicate<const Pred, const_iterator_t<const V>>);
+    STATIC_ASSERT(CanCEnd<const R&> == CanMemberCEnd<const R>);
+    STATIC_ASSERT(!common_range<R>);
+    STATIC_ASSERT(!common_range<const R>);
+    if (!is_empty) {
+        same_as<const_sentinel_t<R>> auto s        = r.cend();
+        same_as<const_sentinel_t<const R>> auto sc = as_const(r).cend();
+
+        if (forward_range<V>) { // intentionally not if constexpr
+            // Compare with const / non-const iterators
+            assert(s != r.cbegin());
+            assert(s != as_const(r).cbegin());
+            assert(sc != r.cbegin());
+            assert(sc != as_const(r).cbegin());
+
+            // Compare with end of range
+            if constexpr (common_range<V>) {
+                assert(s == s.base());
+                assert(s == sc.base());
+                assert(sc == s.base());
+                assert(sc == sc.base());
+            } else if constexpr (forward_range<V> && is_lvalue_reference_v<Rng>) {
+                auto full_range   = views::take_while(rng, [](auto const&) { return true; });

```

Comment:
This lambda should say `const auto&` for consistency. (Copied from a pre-existing occurrence.)